File: src/main/java/br/com/zumbolovsky/fateapp/service/TestBProcessorService.java
package br.com.zumbolovsky.fateapp.service;

import org.springframework.stereotype.Component;

@Component
public class TestBProcessorService implements TestProcessorService {

    @Override
    public String process() {
        return "I'm implementation B";
    }

    @Override
    public boolean supports(TestEnum delimiter) {
        return TestEnum.TEST_B == delimiter;
    }
}

File: src/main/kotlin/br/com/zumbolovsky/fateapp/service/TestBProcessorService.kt
package br.com.zumbolovsky.fateapp.service

import org.springframework.stereotype.Component

@Component
class TestBProcessorService : TestProcessorService {
    override fun process(): String {
        return "I'm implementation B"
    }

    override fun supports(delimiter: TestEnum): Boolean {
        return TestEnum.TEST_B == delimiter
    }
}